[DOC-REVIEW] Nuclio-contribution doc review (#1327)
nuclio/nuclio: docs/devel/contributing.md

This guide will guide you through the process of setting up a development environment and contributing to Nuclio. 

#### In This Document
- [Prerequisites](#prerequisites)
- [Getting the source code](#get-source)
- [Setting up a GoLand project](#goland-setup)
- [Submitting a PR](#submitting-a-pr)

<a id="prerequisites"></a>
## Prerequisites

Ensure that your setup includes the following prerequisite components:

- Linux or OSX

> - Git

Reviewer: ```suggestion
- git
```

mainly because it refers to the command line utility

Author: It's capitalized in the official Git documentation — https://git-scm.com/.

- Docker version 17.05+ (because Nuclio uses [multi-stage builds](https://docs.docker.com/engine/userguide/eng-image/multistage-build/))
- The Go toolchain (CI tests with 1.10, best use that)
- A **GOPATH** directory and a `GOPATH` environment variable that's set to that directory
- Kubernetes version 1.7+ (mostly for testing) - `minikube` recommended; (you can follow the [Minikube getting-started guide](/docs/setup/minikube/getting-started-minikube.md))

<a id="get-source"></a>
## Getting the source code

Fork the Nuclio GitHub repository and clone it:

```sh
git clone https://github.com/<your username>/nuclio.git $GOPATH/src/github.com/nuclio/nuclio
```

Check out the `development` branch, because that's where all the goodness happens:

```sh
cd $GOPATH/src/github.com/nuclio/nuclio && git checkout development
```

Now, use `go get` to get some dependencies that are injected into functions and cannot be vendored:

```sh
go get github.com/v3io/v3io-go-http/... && go get github.com/nuclio/logger/... && go get github.com/nuclio/nuclio-sdk-go/... && go get github.com/nuclio/amqp/... && go get github.com/v3io/scaler-types/...
```

> Build Nuclio artifacts (`nuctl`, container images):

Reviewer: what is wrong with Docker images?

Author: Both terms are OK, but we wanted to unify our terminology (which was inconsistent) and in consultation with Yaron H. and Oded we decided to use "container images" — which is more generic and isn't restricted to Docker; (currently, we only supported Docker as a container platform, but this might change in the future). We explain in the intro that we support Docker. See PR #1326.

```sh
make build
```

You should now have quite a few `nuclio/<something>` images tagged as `latest-amd64`, along with `nuctl-latest-<os>-amd64` with a `nuctl` symbolic link under `$GOPATH/bin`. Now, run a few tests:

```sh
make lint test
```

> This may take a while (about 10 minutes) and requires only Docker. End to end testing on Kubernetes is still done manually.

Reviewer: ```suggestion
This could take up to 10 minutes and requires only Docker. integration testing on Kubernetes is still done manually.
```
